fsspec/callbacks.py

```diff
@@ -25,6 +28,56 @@ def __init__(self, size=None, value=0, hooks=None, **kwargs):
         self.hooks = hooks or {}
         self.kw = kwargs
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *exc_args):
+        self.close()
+
+    def close(self):
+        """Close callback."""
+
+    def branched(self, path_1, path_2, kwargs):
```

Comment:
Here, I matched the function signature with `branch`, but I am open to any suggestions here (if there should be `kwargs` or `**kwargs` or if it should be there at all).

Reply:
I have no better suggestion.

Reply:
I am trying to think of what could be the use case of `kwargs`. So far, it feels like an unnecessary required argument to pass.

I think following will be an example of how someone will use a `branched` callback. 

```python
with callback.branched(path_1, path_2, {}) as child:
    pass
```

The kwargs feel unnecessary to me. Either it should be `kwargs=None` by default or should be removed. I am leaning towards removing it though (can be added if we get some requests).